adafruit_onewire/bus.py

```diff
@@ -0,0 +1,206 @@
+        return self._rom[7]
+
+    @property
+    def serial_number(self):
+        """The 48 bit serial number."""
+        return self._rom[1:7]
+
+    @property
+    def family_code(self):
+        """The 8 bit family code."""
+        return self._rom[0]
+
+class OneWireBus(object):
+    """A class to represent a 1-Wire bus."""
+
+    def __init__(self, pin):
+        self._ow = busio.OneWire(pin)
+
+    def reset(self, required=False):
+        """
+        Perform a reset and check for presence pulse.
+
+        :param bool required: require presence pulse
+        """
+        reset = self._ow.reset()
+        if required and reset:
+            raise OneWireError
+        return not reset
+
+    def readinto(self, buf, **kwargs):
+        """
+        Read into ``buf`` from the device. The number of bytes read will be the
+        length of ``buf``.
+
+        If ``start`` or ``end`` is provided, then the buffer will be sliced
+        as if ``buf[start:end]``. This will not cause an allocation like
+        ``buf[start:end]`` will so it saves memory.
+
+        :param bytearray buf: buffer to write into
+        :param int start: Index to start writing at
+        :param int end: Index to write up to but not include
+        """
+        start = 0
+        end = len(buf)
+        if 'start' in kwargs:
+            start = kwargs['start']
+        if 'end' in kwargs:
+            end = kwargs['end']
+        for i in range(start, end):
+            buf[i] = self._readbyte()
+
+    def write(self, buf, **kwargs):
+        """
+        Write the bytes from ``buf`` to the device.
+
+        If ``start`` or ``end`` is provided, then the buffer will be sliced
+        as if ``buffer[start:end]``. This will not cause an allocation like
+        ``buffer[start:end]`` will so it saves memory.
+
+        :param bytearray buf: buffer containing the bytes to write
+        :param int start: Index to start writing from
+        :param int end: Index to read up to but not include
+        """
+        start = 0
+        end = len(buf)
+        if 'start' in kwargs:
+            start = kwargs['start']
+        if 'end' in kwargs:
+            end = kwargs['end']
+        for i in range(start, end):
+            self._writebyte(buf[i])
+
+    def scan(self):
+        """Scan for devices on the bus and return a list of addresses."""
+        devices = []
+        diff = 65
+        rom = False
+        for i in range(0xff): # pylint: disable=unused-variable
+            rom, diff = self._search_rom(rom, diff)
+            if rom:
+                devices += [OneWireAddress(rom)]
+            if diff == 0:
+                break
+        return devices
+
+    def _readbit(self):
+        return self._ow.read_bit()
+
+    def _readbyte(self):
+        val = 0
+        for i in range(8):
+            val |= self._ow.read_bit() << i
+        return val
+
+    def _writebit(self, value):
+        return self._ow.write_bit(value)
```

Comment:
Same here. By the way, are those even used anywhere?

Reply:
read_bit is used to check for conversion completion

Reply:
oops. they were used by that. not any more. they are still used by _search_rom though.